pkg/manifestclient/read_roundtripper.go

```diff
@@ -128,10 +140,94 @@ func isServerGroupResourceDiscovery(path string) bool {
 	if path == "/api/v1" {
 		return true
 	}
+	if path == "/api" {
+		return true
+	}
 
 	parts := strings.Split(path, "/")
 	if len(parts) != 4 {
 		return false
 	}
 	return parts[0] == "" && parts[1] == "apis"
 }
+
+//go:embed default-discovery
+var defaultDiscovery embed.FS
+
+func (mrt *manifestRoundTripper) getKindForResource(gvr schema.GroupVersionResource) (kindData, error) {
```

Comment:
`kindData` leaks the `err` filed which is not used outside of this function, can we get that fixed?

Reply:
> `kindData` leaks the `err` filed which is not used outside of this function, can we get that fixed?

How would you cache the error with out storing it in kindData?  We hit this path thousands of times.